File: scytale/static/js/app/mixed.js
(function($, APP) {
    APP.mixed = function(selector, alphabet) {
        var update = function() {
            APP.clean($(selector), alphabet);
            APP.encrypt(
                $(".plaintext pre").text(),
                $(selector).val(),
                $(".ciphertext pre"));
        };
        $(selector).on("change input", update);
        update();
    };
})(jQuery, APP || {});

File: scripts/encrypt/mixed.py
import random

from scytale.ciphers import MixedAlphabet


if __name__ == '__main__':
    plaintext = input('Plaintext: ')
    key = input('Key: ')
    if not key:
        a = list(MixedAlphabet.alphabet)
        random.shuffle(a)
        key = ''.join(a)
        print(f'Random Key: {key}')
    ciphertext = MixedAlphabet(key=key).encrypt(plaintext)
    print(f'Ciphertext: {ciphertext}')
